internal/webhook/multiclusterservice_webhook.go

```diff
@@ -0,0 +1,109 @@
+
+	"github.com/Mirantis/hmc/api/v1alpha1"
+	"github.com/Mirantis/hmc/internal/utils"
+)
+
+type MultiClusterServiceValidator struct {
+	client.Client
+}
+
+const invalidMultiClusterServiceMsg = "the MultiClusterService is invalid"
+
+// SetupWebhookWithManager will setup the manager to manage the webhooks
+func (v *MultiClusterServiceValidator) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	v.Client = mgr.GetClient()
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&v1alpha1.MultiClusterService{}).
+		WithValidator(v).
+		WithDefaulter(v).
+		Complete()
+}
+
+var (
+	_ webhook.CustomValidator = &MultiClusterServiceValidator{}
+	_ webhook.CustomDefaulter = &MultiClusterServiceValidator{}
+)
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (*MultiClusterServiceValidator) Default(_ context.Context, _ runtime.Object) error {
+	return nil
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
+func (v *MultiClusterServiceValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	mcs, ok := obj.(*v1alpha1.MultiClusterService)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected MultiClusterService but got a %T", obj))
+	}
+
+	if err := validateServices(ctx, v.Client, utils.DefaultSystemNamespace, mcs.Spec.Services); err != nil {
+		return nil, fmt.Errorf("%s: %w", invalidMultiClusterServiceMsg, err)
+	}
+
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+func (v *MultiClusterServiceValidator) ValidateUpdate(ctx context.Context, _, newObj runtime.Object) (admission.Warnings, error) {
+	mcs, ok := newObj.(*v1alpha1.MultiClusterService)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected MultiClusterService but got a %T", newObj))
+	}
+
+	if err := validateServices(ctx, v.Client, utils.DefaultSystemNamespace, mcs.Spec.Services); err != nil {
+		return nil, fmt.Errorf("%s: %w", invalidMultiClusterServiceMsg, err)
+	}
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
+func (*MultiClusterServiceValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+	return nil, nil
+}
+
+func getServiceTemplate(ctx context.Context, c client.Client, templateNamespace, templateName string) (tpl *v1alpha1.ServiceTemplate, err error) {
+	tpl = new(v1alpha1.ServiceTemplate)
+	return tpl, c.Get(ctx, client.ObjectKey{Namespace: templateNamespace, Name: templateName}, tpl)
+}
+
+func validateServices(ctx context.Context, c client.Client, namespace string, services []v1alpha1.ServiceSpec) error {
+	for _, svc := range services {
+		tpl, err := getServiceTemplate(ctx, c, namespace, svc.Template)
+		if err != nil {
+			return err
+		}
+
+		if err := isTemplateValid(tpl.GetCommonStatus()); err != nil {
```

Comment:
I think it's better to collect all the errors (in case there is more than one invalid template) and then return all of the errors (to have a clearer validation message). But it's up to you.